### PR TITLE
Crossdock errors testing

### PR DESCRIPTION
**Pull request by junchaowu**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -5,44 +5,46 @@ services:
         links:
             - java
             - node
-            - go
-            - python
-            - python-sync
+#            - go
+#            - python
+#            - python-sync
```
Comment (breerly) [inline docker-compose.yml]: Can you uncomment these? Behaviors that aren't implemented should just be skipped, they shouldn't fail the test suite.

File: docker-compose.yml [hunk 2]
```diff
@@ -5,44 +5,46 @@ services:
...
             - AXIS_TRANSPORT=http,tchannel
-
-            - BEHAVIOR_RAW=client,server,transport
-            - BEHAVIOR_JSON=client,server,transport
-            - BEHAVIOR_THRIFT=client,server,transport
+            - AXIS_ERRORS=node
+#
+#            - BEHAVIOR_RAW=client,server,transport
+#            - BEHAVIOR_JSON=client,server,transport
+#            - BEHAVIOR_THRIFT=client,server,transport
```
Comment (breerly) [inline docker-compose.yml]: lets make sure these are uncommented before landing.

This feature will be useful for keeping these uncommented: https://github.com/yarpc/crossdock/issues/44

File: docker-compose.yml (hunk 1 above)
Comment (junchaowu) [inline docker-compose.yml]: can't do it right. The go and python will fail on crossdock test

**Comment by abhinav** [comment]
Just an update: You can test this against `yarpc-{go,node}:crossdock-errors` now. You have a few minor things to fix:

The error you raise for `unexpected-error` is wrong.

- Go

  ```
  Error: Not equal: "UnexpectedError: error for procedure \"unexpected-error\" of service \"yarpc-test\": error\n" (expected)
          != "UnexpectedError: error for procedure \"unexpected-error\" of service \"yarpc-test\": bad boy\n" (actual)
  Messages: unexpected error: response body should be informative error
  ```

- Node

  ```
  expected: 'UnexpectedError: error for procedure "unexpected-error" of service "yarpc-test": error\n'
  actual: 'UnexpectedError: error for procedure "unexpected-error" of service "yarpc-test": bad boy\n'
  ```

Also, calls from Java to Node, Go, and Java for the `errors` behavior simply
fail:

```
F - {0xc820013020 java map[behavior:errors server:go]} - err: net/http: request canceled
F - {0xc820013020 java map[behavior:errors server:node]} - err: net/http: request canceled
F - {0xc820013020 java map[behavior:errors server:java]} - err: net/http: request canceled
```

**Comment by junchaowu** [comment]
The 'bad body' is the actually error message I throw in the function. Java and node test pass before they should only check the template part.

**Comment by junchaowu** [comment]
errors behaviors java is the server , only node will be the client. because node implements a bad node client which will send out bad requests . We should only use node as client for error behaviors.

**Comment by abhinav** [comment]
Whether we use just Node or Node and Go for the test client is not really relevant to this discrepancy. The behavior of the test subjects (the RPC server implementations) has to be the same in all languages. yarpc-node's implementation of the 'unexpected-error' procedure on the test subject is here: https://github.com/yarpc/yarpc-node/blob/crossdock-errors/test/subject.js#L188

```javascript
    this.json.register({
        procedure: 'unexpected-error'
    }, function error(request, callback) {
        callback(new Error('error'));
    });
```

So your implementation of the test subject should return the same error and message. We'll want to update the [test-subject.md](https://github.com/yarpc/yarpc/blob/master/test-subject.md) with this as well: The test subject must implement an `unexpected-error` procedure that fails with an unexpected error with the message "error".

**Comment by junchaowu** [comment]
sure l'll update that.

**Comment by junchaowu** [comment]
this PR is deprecated because the middle change will have latest error handling.

**Pull request closed without merging**